plutus-conformance/src/PlutusConformance/Common.hs

```diff
@@ -0,0 +1,134 @@
+{-# LANGUAGE OverloadedStrings #-}
+
+{- | Plutus conformance test suite library. -}
+module PlutusConformance.Common where
+
+import Control.Exception (SomeException, evaluate, try)
+import Data.Text qualified as T
+import Data.Text.IO qualified as T
+import PlutusCore.Core (defaultVersion)
+import PlutusCore.Default (DefaultFun, DefaultUni)
+import PlutusCore.Error (ParserErrorBundle (ParseErrorB))
+import PlutusCore.Evaluation.Machine.ExBudgetingDefaults (defaultCekParameters)
+import PlutusCore.Evaluation.Result (EvaluationResult (..))
+import PlutusCore.Name (Name)
+import PlutusCore.Quote (runQuoteT)
+import PlutusPrelude
+import Test.Tasty (TestTree, defaultMain, testGroup)
+import Test.Tasty.Golden (findByExtension)
+import Test.Tasty.HUnit (testCase, (@=?))
+import Text.Megaparsec (SourcePos)
+import UntypedPlutusCore.Core.Type qualified as UPLC
+import UntypedPlutusCore.Evaluation.Machine.Cek (unsafeEvaluateCekNoEmit)
+import UntypedPlutusCore.Parser qualified as UPLC
+
+-- | A TestContent contains what you need to run a test.
+data TestContent =
```

Comment:
I think we can use `TestContent` for all the tests? I'll be restructuring quite a bit given your other comments, we'll see how that turns out. But I do want to have as much common stuff as I can to minimize code repetition.

Reply:
I still think it's good to keep `TestContent` as is for all tests. In the future I may create a data type that includes all the info you need for a type of test, like 

```haskell
data TestTypes =
-- `String` is the test group name 
-- FilePath is the dir
  UplcEvalTests TestContents String FilePath Runner
  PLCTCTests ...
```
And then I can make a few of the functions more abstract/generic. But I will do it later on as I add more test types.

Reply:
I may not understand fully but atm I think we're good without using `IsTest`. Lmk what you think.

Reply:
I still think it would be nicer, but I think that for kind of abstract reasons, namely that it's a closer modelling of the domain.

My understanding of what we're doing (in English) is something like:
1. We look at all the test case specifications, and for each one we make a test case
2. Then we give all the test cases to the test runner in a structured way, and the test runner runs them

`tasty`'s type for a test case is `TestTree`, so `mkUplcTestCase` does part 1, which is good. But it's the argument to `mkUplcTestCase` that feels off to me. Because it needs two bits to specify the test case, namely the runner function and the file paths. It seems to me like those together specify the test case, so it makes sense to bundle them. You also then have a natural unit of encapsulation, namely the tests specification type and the functions for running the test, etc.

This then _fits_ with `tasty`'s idea of a test case provider, which is a precisely a datatype that contains the specification of a test and knows how to run it. So it all lines up nicely, and we get extra things which we might not want now but could be useful later e.g. the ability to pass custom options to tests. 

Anyway, I think we should merge this as it is, and maybe I'll try the refactoring and see what you think of it.